KilianG98-A2

KilianG98 opened the pull request:
Ready for review:)

DriftKing1998 commented:
Hi Kilian,

I downloaded and tested your code. It did work fine for the first input, had problems with the second input.
In the beginning I thought it had a problem, but by printing the 'listOfPaths' each time I saw that the solutions changed over time. I think you chose the DeapthFirst-Route, but since your code had two loops which loop through each 'town' in every iteration, it takes a really long time. After ~30 min I stopped the algorithm. 

You should maybe remove the already chosen 'townes' from the possibilities for this one solution, so the algorithm does not have to loop through the entire set each time. I saved each potential solution as a tuple of the remaining 'townes' and the already connected 'townes'. Then I do not have to loop through each 'town' everytime I searched for a new tuple.

Apart from that I liked your decoding of the input. It was really straightforward. but I think the fact that you used pd.columns for your loops put you on a bad the track. Your argument-parser also worked fine and was clean. Furthermore, I liked the use of 'try and expect' statements at some points to ensure correct results.

Hofer Michael

KilianG98 commented:
> Hi Kilian,
> 
> I downloaded and tested your code. It did work fine for the first input, had problems with the second input. In the beginning I thought it had a problem, but by printing the 'listOfPaths' each time I saw that the solutions changed over time. I think you chose the DeapthFirst-Route, but since your code had two loops which loop through each 'town' in every iteration, it takes a really long time. After ~30 min I stopped the algorithm.
> 
> You should maybe remove the already chosen 'townes' from the possibilities for this one solution, so the algorithm does not have to loop through the entire set each time. I saved each potential solution as a tuple of the remaining 'townes' and the already connected 'townes'. Then I do not have to loop through each 'town' everytime I searched for a new tuple.
> 
> Apart from that I liked your decoding of the input. It was really straightforward. but I think the fact that you used pd.columns for your loops put you on a bad the track. Your argument-parser also worked fine and was clean. Furthermore, I liked the use of 'try and expect' statements at some points to ensure correct results.
> 
> Hofer Michael

Hello Michael, 

thank you for your thorough review! Unfortunately i realized too late that my code wasnt working properly and didnt  have time to fix it anymore. For some reason I thought I had tested it on both inputfiles. Your solution sounds like it would speed up the process, I will have to take a look at your implementation in the morning:)

KilianG98 commented:
I implemented the fix you suggested and now I don't have to loop through all cities every time. I also changed my "exploredPaths" variable to a set, which I think gave the main increase in performance. The code is still slow, but it works and gives a result in ~20 seconds.
